types/react/index.d.ts

```diff
@@ -264,6 +265,24 @@ declare namespace React {
         props?: Q, // should be Q & Attributes
         ...children: ReactNode[]): ReactElement<P>;
 
+    // Context via RenderProps
+    type Provider<T> = ComponentType<{
+        value: T,
+        children?: ReactNode
+    }>;
```

Comment:
I'm not sure these should be `ComponentType`; similarly to a `Fragment`, they can't actually be constructed, don't have a `propTypes` property, etc.

Reply:
Why do you say they can't be constructed?  That's exactly how they are used (or perhaps I'm misunderstanding what you mean by "constructed").

Reply:
By "constructed" I mean given as argument to `new`. They're not even functions, as you can see [in their definition](https://github.com/facebook/react/blob/59b39056d91787f6a3e4e0dfc0825c8687bd0af9/packages/react/src/ReactContext.js), but unfortunately TSC doesn't support that 😞

I guess we're stuck with `ComponentType` here, at least until TSC is updated to use the overloads of `React.createElement` for JSX type checking.

https://github.com/Microsoft/TypeScript/issues/21699